### ISSUE TITLE
About heatmap loss

### ISSUE THREAD

**Issue by miooooooooooo** [xingyizhou/CenterNet#670]
When I train hm_loss in my MOT data, I  found  the hm_loss can't converge when use focal loss:loss = - (pos_loss + neg_loss) / num_pos,after several iterations of training,all output heatmaps' value are 1e-4 , because output = torch.clamp(output.sigmoid_(), min=1e-4, max=1-1e-4),it seems all the values are decreasing.I guess it's the imbalance between positive and negative samples.So I change the focal loss: loss = pos_loss / num_pos + neg_loss / neg_pos,and it converges.Can you tell me the reason.

**Comment by xingyizhou** [comment]
I haven't observed this when training on MOT dataset. Can you specify your input resolution and learning rate?

**Comment by miooooooooooo** [comment]
My input resolution is 896*896,and about the learning rate I have tried 1e-4 1e-3 1e-2,but the problem is always exist

**Comment by xingyizhou** [comment]
I usually try a lower learning rate when the loss gives unexpected behavior. I haven't made CenterNet work using learning rate 1e-3 with Adam (You can use SGD with a larger learning rate, but Adam is safer). Also, if you are referring to the MOT benchmark, you can refer to our setting in [CenterTrack](https://github.com/xingyizhou/CenterTrack).

**Comment by miooooooooooo** [comment]
It seems that because I used SGD，I will try it with Adam.Thank you for your reply

**Comment by tengerye** [comment]
@miooooooooooo Hi, if possible, would you please share the result of your attempts? Is it better?

**Comment by noahzhy** [comment]
God, it works to me. I guess its because of single class.

**Comment by Douglassssss** [comment]
> God, it works to me. I guess its because of single class.

hi, @noahzhy , i want to know what's the meaning of 'single class' in your comment. Do you mean your dataset has only one class? Thanks!
